collect_app/src/main/java/org/odk/collect/android/activities/InstanceChooserList.java

```diff
@@ -68,7 +68,7 @@ public void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.chooser_list_layout);
 
         String order;
-        if (getIntent().getStringExtra(ApplicationConstants.BundleKeys.FORM_MODE).equalsIgnoreCase(ApplicationConstants.FormModes.EDIT_SAVED)) {
+        if ((ApplicationConstants.FormModes.EDIT_SAVED).equalsIgnoreCase(getIntent().getStringExtra(ApplicationConstants.BundleKeys.FORM_MODE))) {
```

Comment:
I made these changes because otherwise there was a NullPointerException in InstanceChooserListTest.

The main difference between CONSTANT.equals(object) and object.equals(CONSTANT) is that in is not needed to do null check in the first case, because if object is null we will just have false result of comparison :) The alternative to it can be something like 
if (object != null && object.equals(CONSTANT)). Can you please give your opinion what is the better way here?